npm/private/pnpm.bzl

```diff
@@ -97,59 +73,65 @@ def _v9_resolve_link_version(packages, snapshot_key, name, link):
 def _convert_pnpm_v9_package_dependency_version(packages, snapshots, snapshot_key, name, version):
     if version.startswith("link:"):
         # Resolve link: deps to be workspace root relative
-        version = "link:" + _v9_resolve_link_version(packages, snapshot_key, name, version[5:])
+        return "link:{}|{}".format(name, _v9_resolve_link_version(packages, snapshot_key, name, version[5:]))
```

Comment:
This is a bit odd...
1. we are no longer assuming `{name}@{version}` is unique, so this unique `link:{project-path}` gets carried all the way through
2. carrying this `link:` all the way through instead of switching to `{name}@0.0.0` means we lose the `{name}`
3. ... so I've inserted the name here as `link:{name}|{project-path}`